### ISSUE TITLE
Add skipManyInterleavedTill combinator

### ISSUE THREAD

**Issue by alanz** [bubba/lsp-test#53]
At the moment I have the completion tests in my local environment failing for `hie`, due to

```haskell
_ <- skipManyTill loggingNotification (count 2 noDiagnostics)
```

which in my case receives a diagnostics notification, then a logging notification, then the second diagnostic.

What I actually want to say is skip any logging notifications until we have received two diagnostics notifications.

Is this currently possible? If not, we should make it so.

**Comment by alanz** [comment]
I guess it can be emulated by

```haskell
    _ <- count 2 $ skipManyTill loggingNotification (count 1 noDiagnostics)
```

so perhaps does not need to be anything special. But it does come up in a lot of tests.

**Comment by bubba** [comment]
I think that can be shortened to

```haskell
_ <- count 2 $ skipManyTill loggingNotification noDiagnostics
```

But at this point I think it is worthwhile just looking into a config option to just ignore all logging notifications, for practicality's sake

**Comment by bubba** [comment]
Closed in cb75f5731bd7d85b1f1c684cbb2ce90cf82a1e34
